src/Symfony/Component/Translation/Bridge/Phrase/PhraseProvider.php

```diff
@@ -0,0 +1,305 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Bridge\Phrase;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Mime\Part\DataPart;
+use Symfony\Component\Mime\Part\Multipart\FormDataPart;
+use Symfony\Component\Translation\Bridge\Phrase\Cache\PhraseCachedResponse;
+use Symfony\Component\Translation\Bridge\Phrase\Config\ReadConfig;
+use Symfony\Component\Translation\Bridge\Phrase\Config\WriteConfig;
+use Symfony\Component\Translation\Bridge\Phrase\Event\PhraseReadEvent;
+use Symfony\Component\Translation\Bridge\Phrase\Event\PhraseWriteEvent;
+use Symfony\Component\Translation\Dumper\XliffFileDumper;
+use Symfony\Component\Translation\Exception\ProviderException;
+use Symfony\Component\Translation\Loader\LoaderInterface;
+use Symfony\Component\Translation\Provider\ProviderInterface;
+use Symfony\Component\Translation\TranslatorBag;
+use Symfony\Component\Translation\TranslatorBagInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+/**
+ * @psalm-import-type PhraseReadConfig from ReadConfig
+ * @psalm-import-type PhraseWriteConfig from WriteConfig
+ *
+ * @psalm-type PhraseLocale array{id: string, name: string, code: string, fallback_locale?: ?array{id: string, name: string, code: string}}
+ *
+ * @author wicliff <wicliff.wolda@gmail.com>
+ */
+class PhraseProvider implements ProviderInterface
+{
+    /**
+     * @var array<string, PhraseLocale>
+     */
+    private static array $phraseLocales = [];
+
+    public function __construct(
+        private readonly HttpClientInterface $httpClient,
+        private readonly LoggerInterface $logger,
+        private readonly LoaderInterface $loader,
+        private readonly XliffFileDumper $xliffFileDumper,
+        private readonly EventDispatcherInterface $dispatcher,
+        private readonly CacheItemPoolInterface $cache,
+        private readonly string $defaultLocale,
+        private readonly string $endpoint,
+        private readonly ReadConfig $readConfig,
+        private readonly WriteConfig $writeConfig,
+    ) {
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('phrase://%s', $this->endpoint);
+    }
+
+    /**
+     * @param array<array-key, mixed> $domains
+     * @param array<array-key, mixed> $locales
```

Comment:
I'd remove these: this should be on the parent if worth it + they don't bring much IMHO